projects/flosportsinc/ng-lazy-load/common/lazy-load.directive.ts

```diff
@@ -0,0 +1,137 @@
+import {
+  Directive,
+  Inject,
+  Input,
+  TemplateRef,
+  ViewContainerRef
+} from '@angular/core'
+import { maybe } from 'typescript-monads'
+import { ErrorLogFn } from './lazy-load.interface'
+import { FLO_LAZY_LOAD_LOG_ERROR } from './lazy-load.tokens'
+
+const DEFAULT_THRESHOLD = 0.0
+
+const identity = <A>(a: A): A => a
+
+/**
+ * Lazy load an element by waiting until a trigger element is visible.
+ *
+ * Leverages IntersectionObservers as a browser-native mechanism for lazily
+ * loading DOM content. Once the content is loaded, it will not be unloaded at
+ * any point, as the observer will be destroyed. Be aware that this directive
+ * doesn't produce a placeholder container for the content. If the content can
+ * load in at a position which may move content, a placeholder should be used.
+ *
+ * This directive may not operate as expected without AOT compilation. It
+ * appears that this may not be an issue in Angular 9. Run your development
+ * compilation with AOT to ensure that lazy loading occurs when desired.
+ *
+ * @example
+ * <span #loadTrigger></span>
+ * <app-expensive-component *floLazyLoad="true; threshold=0.2; trigger=loadTrigger;">
+ * </app-expensive-component>
+ *
+ * @see https://caniuse.com/#search=intersectionobserver
+ * @see https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API
+ */
+@Directive({
+  selector: '[libLazyLoad]'
+})
+export class FloLazyLoadDirective {
+  /**
+   * The proportion of the `trigger` element which must be visible to trigger a load.
+   *
+   * Must be at least 0.0 and at most 1.0 but isn't necessary when using
+   * "single-pixel" triggers. It's more useful when trying to load after
+   * scrolling some portion of the way through a larger preceding component.
+   */
+  @Input('libLazyLoadThreshold') public readonly threshold = DEFAULT_THRESHOLD
+
+  /**
+   * The trigger element which will cause the load to occur.
+   *
+   * Remember that this element must be visible on page, otherwise the load will
+   * never occur. An existing element can be used, or an arbitrary element can
+   * be placed on the page as a trigger. The trigger must *always* be present
+   * in the DOM (it cannot target an element with or in \*ngIf or \*ngFor).
+   */
+  @Input('libLazyLoadTrigger') public readonly trigger: HTMLElement
```

Comment:
You can move these to the constructor and give it `@Inject()` values to give it 3 levels of configuration.

Reply:
Can it have both, or does it not need it? e.g.

```typescript
public constructor(
   @Input() @Inject() public readonly trigger: HTMLElement
)
```